api/lib/opentelemetry/instrumentation/caller_locations.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    # The instrumentation CallerLocations contains a reusable function to add caller information to instrumentations
+    class CallerLocations
+      def self.detect_caller_location
+        if ENV['OTEL_EXPORTER_ADD_CALLER_LOCATION']
+          cleaned_up_trace = caller_locations.delete_if {|loc| loc.absolute_path.match('/.rvm/gems/')}
```

Comment:
I'm not too sure what is the smarter/better way to do this, but I thought `'/.rvm/gems/'` could be an ENV var as well.